azerothcore/azerothcore-wotlk
# feat(Core/ObjectMgr): Convert remaining `playercreateinfo_*` tables to use bitmasked values.

## Changes Proposed:
- Changes the initial loading of values from the `playercreateinfo`, `playercreateinfo_action`, and `playercreateinfo_item` to act as bitmasks.
- Changes datatypes and names of various columns in the tables to allow for the change as well as change to upper camel case capitalization.
- Minimal condensing of the values in `playercreateinfo`, converting values in `playercreateinfo_action` and `playercreateinfo_item` to bitmasked values.

## Issues Addressed:

- Closes #3113

## SOURCE:

N/A

## Tests Performed:

- Tested that worldserver still loaded, made a few basic characters to make sure values were being loaded.

## How to Test the Changes:

1. Create characters of various races/classes.
2. Make sure that nothing has changed about your spawn position, spells on your hotbars, or items you're spawned with.

## Known Issues and TODO List:

- There's definitely a better way to convert `playercreateinfo_action` to bitmasks, either some update to the existing values to convert the integers to the values directly, or to insert the condensed entries similar to `playercreateinfo`.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
